tests/unit_tests/restruct_poc/test_rotating_beam.cpp

```diff
@@ -628,4 +632,104 @@ TEST(RotatingBeamTest, CylindricalConstraint) {
     );
 }
 
+TEST(RotatingBeamTest, GeneratorTorque) {
```

Comment:
Similar to the test at the top of this file, it would be good to add a check on the state after the 10 steps are complete to ensure that the results aren't changing as we change the code in the future.